com/metacube/DSA/Assignment5/Dictionary.java

```diff
@@ -0,0 +1,152 @@
+
+
+import java.io.FileNotFoundException;
+import java.io.FileReader;
+import java.io.IOException;
+import java.util.Map;
+import java.util.TreeMap;
+
+import org.json.simple.JSONObject;
+import org.json.simple.parser.JSONParser;
+import org.json.simple.parser.ParseException;
+
+public class Dictionary implements IDictionary {
+	
+	private Map<String, String> dictionary = new TreeMap<String, String>();
+
+	public Map<String, String> getDictionary() {
+		return dictionary;
+	}
+
+	public Dictionary() {
+		// TODO Auto-generated constructor stub
+	}
+	
+	/**
+	 * Constructor to read input from json file
+	 * @param inputDictionary is dictionary file in json
+	 * @throws IOException
+	 * @throws ParseException
+	 * @throws CustomException
+	 */
+	public Dictionary(String inputDictionary) throws IOException, ParseException, CustomException {
+		
+		if(inputDictionary == null) {
+			throw new CustomException("Null Path given.");
+		}
+		
+		JSONParser parser = new JSONParser();
+		try
+        {
+            Object object = parser.parse(new FileReader(inputDictionary));
+            
+            JSONObject jsonObject = (JSONObject)object;
+			
+			for(Object key: jsonObject.keySet()) {
+				addWord(key.toString(), jsonObject.get(key).toString());
+				
+			}
+
+        }
+        catch(FileNotFoundException fe)
+        {
+            fe.printStackTrace();
+        }
+       
+	}
+
+	@Override
+	public boolean addWord(String word, String meaning) {
+		boolean flag = false;
+		try {
+			if(meaning == null || word == null) {
+				throw new CustomException("Null input found.");
+			}
+			
+			word = word.toLowerCase();
+			meaning = meaning.toLowerCase();
+
+			dictionary.put(word, meaning);
+			
+			if(dictionary.containsKey(word)) {
+				flag = true;
+			}
+		}
+		catch(CustomException error) {
+			
+		}
+		
+		return flag;
+	}
+
+	@Override
+	public boolean deleteWord(String word) {
+		boolean flag = false;
+		try {
+			if(word == null) {
+				throw new CustomException("Null input found.");
+			}
+			
+			word = word.toLowerCase();
+			
+			if(!dictionary.isEmpty() && dictionary.containsKey(word)) {
+				dictionary.remove(word);
+				flag = true;
+			}			
+			if(dictionary.containsKey(word)) {
+				flag = false;
+			}
+		}
+		catch(CustomException error) {
+			System.out.println(error.getMessage());
+		}
+		
+		return flag;
+	}
+
+	@Override
+	public String getMeaning(String word) throws CustomException {
+		word = word.toLowerCase();
+		if(word == null) {
+			throw new CustomException("Null input found.");
+		}
+		if(dictionary.isEmpty()) {
+			throw new CustomException("Dictionary is empty");
+		}
+		if(!dictionary.containsKey(word)) {
+			throw new CustomException("Word not present in dictionary.");
+		}
+		return dictionary.get(word);
+	}
+
+	@Override
+	public Map<String, String> sortedDictionary() throws CustomException {
+		if(dictionary.isEmpty()) {
+			throw new CustomException("Dictionary is empty");
+		}
+		return dictionary;		
```

Comment:
If you are sorting by keys than Treemap is always sorted but if you wish to sort by values than you will have to override the comaprator. Question is not clear enough if you need to sort based on values also. So am leaving it as of now. Its just for your knowledge about TreeMaps in java.